src/components/Button.astro

```diff
@@ -0,0 +1,32 @@
+---
+interface Props {
+    variant?: keyof typeof Variant;
+    size?: keyof typeof Size;
+    icon?: string;
+    disabled?: boolean;
+    label: string;
+    onClick?: () => void;
+}
+
+const style = 'inline-flex items-center justify-center rounded-md border disabled:cursor-not-allowed disabled:bg-primary-100 disabled:text-primary-400';
```

Comment:
I would prefer to have it in line in the html because

1. We don't get anything if it is a js variable since the length is the same
2. I'm not sure how astro abstract it but probably it will move it as a variable instead of a html part
3. We loose tailwind autocomplete

Reply:
Excellet, I'll check it